polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -825,7 +826,29 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				let old_holding = self.holding.clone();
 				let result = Config::TransactionalProcessor::process(|| {
 					let deposited = self.holding.saturating_take(assets);
+					let mut all_failed = true;
+					let mut failed_deposits = Vec::new();
 					for asset in deposited.into_assets_iter() {
+						let asset_result = Config::AssetTransactor::deposit_asset(
+							&asset,
+							&beneficiary,
+							Some(&self.context),
+						);
+						match asset_result {
+							Ok(_) => {
+								all_failed = false;
+							}
+							Err(_err) => {
+								// check if is not enough ED
+							}
+						}
+					}
+
+					if all_failed {
+						return Err(XcmError::FailedToTransactAsset(""))
+					}
```

Comment:
We can simply fall through to always retry any failed ones. On second try, they either all work or not.

```suggestion
```